File: app/src/main/java/zhan/rnadomgirdanimation/refresh/IRefreshAdapter.java
package zhan.rnadomgirdanimation.refresh;

import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public abstract class IRefreshAdapter<T> extends RecyclerView.Adapter {

    protected List<T> mData;

    public void setData(List<T> data) {
        mData = data;
    }

    @Override
    public int getItemCount() {
        return mData == null ? 0 : mData.size();
    }
}

File: app/src/main/java/zhan/rnadomgirdanimation/refresh/IRefreshAdapter.kt
package zhan.rnadomgirdanimation.refresh

import androidx.recyclerview.widget.RecyclerView

abstract class IRefreshAdapter<T> : RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    protected var mData: List<T>? = null

    fun setData(data: List<T>) {
        mData = data
    }

    override fun getItemCount(): Int {
        return if (mData == null) 0 else mData!!.size
    }
}
